### PR TITLE
Master imp tour onboarding ipb

### PR DESCRIPTION
**Pull request by bastien-ipb**
Before this commit, the steps with a run "drag_and_drop" was
showing the element to drag but no info on the dropzone.
The selection of a dropdown item had to be done with a click. If
using Enter or Tab, the step was not validated, eventhough the
result was the same. Same issue, if using any hotkey or validating
form with Enter instead of clicking on the button "Save" for example.
When a user lost the target of the pointer, this one disappear until
the element of the step was visible again.

After this commit, the drops will show the element to drag and
the zone where to drop.
The selection of a dropdown item can be done with Enter or Tab.
The hotkeys and validation though Enter will consume the step.
When an element has been found and pointed but disappear after,
the tour will back step until a step that can find her trigger
in the DOM. Like that the user is a bit less lost.

This changes come in the continuation of the tour refactor and
the change of the default run on a tour step. Before it was
"click", but not anymore, it's just a check. So in the manual
step, the "check step" are just ignored because doesn't need any
user interactions.

task-id: 3922501

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_tour/static/src/tour_service/tour_compilers.js [hunk 1]
```diff
@@ -318,65 +351,99 @@ function setupListeners({
...
                 }
-                if (proceedWith) {
-                    return proceedWith;
+
+                if (subActionIndex < currentSubStep) {
+                    return anchorEl;
                 }
 
-                const { triggerEl, altEl, extraTriggerOkay } = findStepTriggers(tour, step);
+                anchorEl = hoot.queryAll(subAction.anchor).at(0);
```
Comment (kebeclibre) [inline addons/web_tour/static/src/tour_service/tour_compilers.js:379]: queryFirst ?

**Review by kebeclibre — COMMENTED** [review]
this seems very promising, thanks

File: addons/web_tour/static/src/tour_service/tour_compilers.js [hunk 2]
```diff
@@ -308,51 +337,210 @@ function setupListeners({
...
+
+    return actions;
+}
+
+/**
+ * @param {HTMLElement} [element]
+ * @param {RunCommand} [runCommand]
+ * @returns {ConsumeEvent[]}
+ */
+export function getConsumeEventType(element, runCommand) {
```
Comment (pipu-odoo) [inline addons/web_tour/static/src/tour_service/tour_compilers.js:388]: Why export it if it's not used anywhere else?

File: addons/web_tour/static/src/tour_service/tour_compilers.js [hunk 3]
```diff
@@ -308,51 +337,210 @@ function setupListeners({
...
+/**
+ *
+ * @param {TourStep} step
+ * @returns {{
+ *  event: string,
+ *  anchor: string,
+ *  altAnchor: string | undefined,
+ * }[]}
+ */
+export function getSubActions(step) {
```
Comment (pipu-odoo) [inline addons/web_tour/static/src/tour_service/tour_compilers.js:349]: Why export it if it's not used anywhere else?

File: addons/web_tour/static/src/tour_service/tour_compilers.js [hunk 4]
```diff
@@ -520,16 +718,18 @@ export function compileStepAuto(stepIndex, step, options) {
...
         pointer,
         stepDelay,
         keepWatchBrowser,
         showPointerDuration,
-        checkDelay,
         onStepConsummed,
         onTourEnd,
     } = options;
     const currentStepIndex = tourState.get(tour.name, "currentIndex");
+    const stepCompiler = mode === "auto" ? compileStepAuto : compileStepManual;
```
Comment (pipu-odoo) [inline addons/web_tour/static/src/tour_service/tour_compilers.js:730]: Nice !

**Comment by kebeclibre** [comment]
robodoo rebase-ff r+

**Comment by bastien-ipb** [comment]
@odoo/rd-security Hello security team, could you override the rule on this PR ? I replace the old way "window.location" by "browser.location"

**Review by xmo-odoo — COMMENTED** [review]
@robodoo override=ci/security

**Comment by mart-e** [comment]
@bastien-ipb but why is it safe?

**Comment by bastien-ipb** [comment]
> @bastien-ipb but why is it safe?

It's mainly for the tests. When changing location or timeout using browser instead of the global window allow to patch them during the tests.

**Comment by kebeclibre** [comment]
robodoo delagate+

**Comment by kebeclibre** [comment]
robodoo delegate+ rebase-ff

**Comment by bastien-ipb** [comment]
robodoo r+

**Comment by bastien-ipb** [comment]
robodoo r+

**Pull request closed without merging**